src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -137,7 +145,6 @@ ss::future<> remove_client_if(List& list, Pred pred) {
           });
     }
 }
-} // namespace
 
 ss::future<> kafka_client_cache::clean_stale_clients() {
     constexpr auto is_expired = [](std::chrono::milliseconds keep_alive) {
```

Comment:
I think you need a gate and lock here, drop `invoke_on_cache`, always call into `fetch_or_insert` directly..

Reply:
Dropping invoke_on_cache would mean moving the gate and lock into `fetch_of_insert` thus making it async.
IMO, we need to have scheduling mechs on the cache to protect against concurrent calls to `fetch_or_insert`. Or am I missing something here?

Reply:
`fetch_or_insert` doesn't need a gate or lock.